cmd/goal/application.go

```diff
@@ -1123,6 +1121,21 @@ var methodAppCmd = &cobra.Command{
 			reportErrorf(errorSigningTX, err)
 		}
 
+		if outFilename != "" {
```

Comment:
I think this should be moved right before `// Broadcast`, so the transaction does not always get signed. This matches the behavior of the other subcommands.